ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -234,7 +234,7 @@ public final int doPhase2Commit ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
-		if (s != org.omg.CosTransactions.Status.StatusPrepared) {
+		if (s != org.omg.CosTransactions.Status.StatusPrepared && s != org.omg.CosTransactions.Status.StatusCommitted) {
```

Comment:
it is in the case of XA_RETRY on the XAR I need to make sure it gets a commit call, I wonder about bottom-up recovery let me check